### PR TITLE
Issue resolved for contract balance discrepancy

### PR DESCRIPTION
**Pull request by abhishek-k-sharma-lab49**

### CODE REVIEW

File: contracts/Pair.sol [hunk 1]
```diff
@@ -364,6 +397,34 @@ contract Pair is IPair, Initializable {
         return tokenQ;
     }
 
+    function _calculateSwapTokenAQuantities(
+        int256 senderSwapQty
+    ) private view returns (int256, int256, int256) {
+        // Token A Calculation
+        int256 feeTokenA = feeForToken(senderSwapQty);
+        int256 _tokenATreasureFee = feeTokenA / 2; //50% goes to treaasurer
```
Comment (abhishek-k-sharma-lab49) [inline contracts/Pair.sol:405]: typo "treasurer"

File: contracts/Pair.sol [hunk 2]
```diff
@@ -147,47 +147,61 @@ contract Pair is IPair, Initializable {
...
-        int256 feeTokenB = feeForToken(deltaBQty);
-        // deduct fee from the token B
-        deltaBQty = deltaBQty - (feeTokenB / 2);
+
+        // Token A Calculation
+        (
+            int256 _tokenATreasureFee,
+            int256 _deltaAQtyAfterAdjustingFee,
+            int256 _tokenASwapQtyPlusContractTokenShare
+        ) = _calculateSwapTokenAQuantities(_deltaAQty);
```
Comment (abhishek-k-sharma-lab49) [inline contracts/Pair.sol:156]: _calculateSwapTokenAQuantities name could be changed to "_calculateTradeTokenQuantities" as it is being used for TokenB as well.
Similarly _calculateSwapTokenQuantities could be used or any elaborative name.

File: contracts/Pair.sol [hunk 3]
```diff
@@ -364,6 +397,34 @@ contract Pair is IPair, Initializable {
         return tokenQ;
     }
 
+    function _calculateSwapTokenAQuantities(
+        int256 senderSwapQty
+    ) private view returns (int256, int256, int256) {
+        // Token A Calculation
+        int256 feeTokenA = feeForToken(senderSwapQty);
+        int256 _tokenATreasureFee = feeTokenA / 2; //50% goes to treaasurer
+        int256 tokenAContractShare = feeTokenA / 2; //50% goes to contract
```
Comment (abhishek-k-sharma-lab49) [inline contracts/Pair.sol:406]: Seems we can change the naming to tokenToTrade and TokenToGet.

File: contracts/Pair.sol [hunk 4]
```diff
@@ -364,6 +397,34 @@ contract Pair is IPair, Initializable {
...
+                tokenAContractShare;
+
+        return (
+            _tokenATreasureFee,
+            _deltaAQtyAfterAdjustingFee,
+            _tokenASwapQtyPlusContractTokenShare
+        );
+    }
+
+    function _calculateSwapTokenBQuantities(
```
Comment (abhishek-k-sharma-lab49) [inline contracts/Pair.sol:419]: Seems we can change the naming to tokenToTrade and TokenToGet.

File: contracts/Pair.sol (hunk 2 above)
Comment (amandeeprehal-lab49) [reply contracts/Pair.sol:156]: yes, right, let me make it generic

Comment (amandeeprehal-lab49) [reply contracts/Pair.sol]: @abhishek-k-sharma-lab49  addressed.

File: contracts/Pair.sol (hunk 3 above)
Comment (amandeeprehal-lab49) [reply contracts/Pair.sol]: yes, changed now.

File: contracts/Pair.sol (hunk 4 above)
Comment (amandeeprehal-lab49) [reply contracts/Pair.sol]: done

**Review by shivamgarglab49 — APPROVED** [review]

**Pull request merged**